mlflow/llama_index/tracer.py

```diff
@@ -143,11 +145,14 @@ def new_span(
         parent_span_id: Optional[str] = None,
         **kwargs: Any,
     ) -> _LlamaSpan:
+        with self.lock:
+            parent = self.open_spans.get(parent_span_id) if parent_span_id else None
```

Comment:
note: The `open_spans` is inherited from base handler but it does not seem thread safe. While the simple test cases work fine without this locking, this may becomes an annoying issue for complex workflow.

Reply:
No issues with this approach. Good looking out for potential contention :)